pkg/virt-handler/cmd-client/client.go

```diff
@@ -238,9 +238,28 @@ func IsDisconnected(err error) bool {
 	return false
 }
 
-func (c *VirtLauncherClient) SyncVirtualMachine(vmi *v1.VirtualMachineInstance) error {
-	return c.genericSendVMICmd("SyncVMI", c.v1client.SyncVirtualMachine, vmi)
+func (c *VirtLauncherClient) SyncVirtualMachine(vmi *v1.VirtualMachineInstance, options *cmdv1.VirtualMachineOptions) error {
 
+	vmiJson, err := json.Marshal(vmi)
+	if err != nil {
+		return err
+	}
+
+	request := &cmdv1.VMIRequest{
+		Vmi: &cmdv1.VMI{
+			VmiJson: vmiJson,
+		},
+		Options: &cmdv1.VirtualMachineOptions{
```

Comment:
2 things here:
1. You already have the correct options type here now, no need to reconsctruct it
2. We now have some code duplication with genericSendVMICmd: can you please add the options arg to that func, and let the other users of that func just pass nil to it?

Reply:
> We now have some code duplication with genericSendVMICmd: can you please add the options arg to that func, and let the other users of that func just pass nil to it?

It seems I am not the only one who wants to fix it here :) https://github.com/kubevirt/kubevirt/pull/2472#discussion_r310978526 as @vladikr mentioned we should refactor ```MigrateVirtualMachine``` as well.

Reply:
> 2 things here:
> 
> 1. You already have the correct options type here now, no need to reconsctruct it
Corrected.
> 2. We now have some code duplication with genericSendVMICmd: can you please add the options arg to that func, and let the other users of that func just pass nil to it?

Reply:
@mfranczy ha, missed that :D But I would be fine with skipping MigrateVirtualMachine for now, as it uses another request type (MigrationRequest vs VMIRequest)

Reply:
> 2 things here:
> 
> 1. You already have the correct options type here now, no need to reconsctruct it
> 2. We now have some code duplication with genericSendVMICmd: can you please add the options arg to that func, and let the other users of that func just pass nil to it?

@slintes @mfranczy @vladikr 
Not touching Migration atm, but corrected it with an additional arg in genericSendVMICmd now